types/errors.go

```diff
@@ -0,0 +1,15 @@
+package types
+
+import "errors"
+
+var (
+	ErrInvalidNamespaceId          = errors.New("invalid lenght of 'NamespaceId'")
+	ErrInvalidLastHeaderHash       = errors.New("invalid lenght of 'InvalidLastHeaderHash'")
+	ErrInvalidLastCommitHeaderHash = errors.New("invalid lenght of 'InvalidLastCommit.HeaderHash'")
+	ErrInvalidLastCommitHash       = errors.New("invalid lenght of 'InvalidLastCommitHash'")
+	ErrInvalidDataHash             = errors.New("invalid lenght of 'InvalidDataHash'")
+	ErrInvalidConsensusHash        = errors.New("invalid lenght of 'InvalidConsensusHash'")
+	ErrInvalidAppHash              = errors.New("invalid lenght of 'InvalidAppHash'")
+	ErrInvalidLastResultsHash      = errors.New("invalid lenght of 'InvalidLastResultsHash'")
+	ErrInvalidProposerAddress      = errors.New("invalid lenght of 'InvalidProposerAddress'")
```

Comment:
What's the point of making those errors public?

Reply:
Just a sentinel value, if case someone would... expect some particular field (de)serialization error? Well, actually it may not be the most important error value to be exported.

Reply:
I just used to care about things getting exported and those errors seem excessive. Furthermore, most just expect serialization to work and I can't image any case where you would expect some specific field to fail outside the package. Let's keep things minimal. Even more, I would remove those error var at all and just use `fmt.Errorf`, unless there is at least one or two additional usages.

@liamsi, wdut?

Reply:
One use case where I can imagine using those errors are tests (in case of testing field by field, sentinel values will help). But in this situation, I don't need to export the vars.

Reply:
Let's keep them unexported for now. As @Wondertan said, let's be cautious of what we make part of the public API (and these errors are currently part of the public API). This is not too relevant right now but it will become relevant as soon as we have a first stable release. Users will expect stability around the exposed APIs.